.github/workflows/ci.yml

```diff
@@ -180,7 +180,7 @@ jobs:
       if: steps.cache.outputs.cache-hit != 'true'
       run: |
         if [ ${{ matrix.mypy-version }} != 'lockfile' ]; then
-          pdm remove mypy && pdm add mypy==${{ matrix.mypy-version }}
+          pdm remove -G linting mypy && pdm remove -G mypy mypy && pdm add mypy==${{ matrix.mypy-version }}
```

Comment:
I think that would be better but I spent a lot of time fiddling with this and this works so I don't want to go down another rabbit hole of this stuff quite yet. Definitely happy to merge that in a separate PR, or this one if it will definitely work and I don't have to debug CI within CI runs, which is how I had to do it for hours last night